src/server/components/annual-review/controller.ts

```diff
@@ -0,0 +1,145 @@
+import { Request, Response, NextFunction } from "express";
+import { startCase } from "lodash";
+import { Status } from "@prisma/client";
+import { add, isPast } from "date-fns";
+
+import * as Types from "../dashboard/listsItems/types";
+import { findListItemByReference } from "server/models/listItem/listItem";
+import { getDetailsViewModel } from "server/components/dashboard/listsItems/getViewModel";
+import { getCSRFToken } from "server/components/cookies/helpers";
+import { HttpException } from "server/middlewares/error-handlers";
+import { prisma } from "server/models/db/prisma-client";
+
+import type { ListItemGetObject, List } from "server/models/types";
+
+export async function confirmGetController(req: Request, res: Response, next: NextFunction): Promise<void> {
+  try {
+    const { listItemRef } = req.params;
+    const listItem = (await findListItemByReference(listItemRef)) as ListItemGetObject;
+    const rows = formatDataForSummaryRows(listItem);
+    const errorMsg = req.flash("annualReviewError")[0];
+    const userHasConfirmed = listItem.status === Status.CHECK_ANNUAL_REVIEW
+    let error = null;
+
+    if (await dateHasExpired(listItem.id)) {
+      return res.redirect("/annual-review/error?expired=true");
```

Comment:
use PRG + flash instead of query params.

Reply:
Luckily for me this is a get controller so I don't need to do any PRG 😉